cloudflare.py

```diff
@@ -242,14 +242,21 @@ class Zone(object):
 
     def __init__(self, name, zone):
         self.name = name
-        self.auth_email = zone["auth_email"]
-        self.auth_key = zone["auth_key"]
+        self.api_token = zone.get("api_token", None)
+        self.auth_email = zone.get("auth_email", None)
+        self.auth_key = zone.get("auth_key", None)
         self.zone_id = zone["zone_id"]
         self.records = zone["records"]
         self.exclude = zone.get('exclude', [])
+        
+        if not self.api_token and (not self.auth_email or not slef.auth_key):
```

Comment:
You have a typo here :) `slef.auth_key` --> `self.auth_key`

Reply:
I don't know where that comes from...
That mistake is not in the file we use in our env...